src/RequestProcessor.cs

```diff
@@ -144,18 +151,40 @@ internal StreamingMessage ProcessFunctionLoadRequest(StreamingMessage request)
                 out StatusResult status);
             response.FunctionLoadResponse.FunctionId = functionLoadRequest.FunctionId;
 
-            try
+            // When a functionLoadRequest comes in, we check to see if a dependency download has failed in a previous call
+            // or if PowerShell could not be initialized. If this is the case, mark this as a failed request
+            // and submit the exception to the Host (runtime).
+            if (_initTerminatingError != null)
             {
-                // Ideally, the initialization should happen when processing 'WorkerInitRequest', however, the 'WorkerInitRequest'
-                // message doesn't provide the file path of the FunctionApp. That information is not available until the first
-                // 'FunctionLoadRequest' comes in. Therefore, we run initialization here.
-                if (!_isFunctionAppInitialized)
+                status.Status = StatusResult.Types.Status.Failure;
+                status.Exception = _initTerminatingError.ToRpcException();
+                return response;
+            }
+
+            // Ideally, the initialization should happen when processing 'WorkerInitRequest', however, the 'WorkerInitRequest'
+            // message doesn't provide information about the FunctionApp. That information is not available until the first
+            // 'FunctionLoadRequest' comes in. Therefore, we run initialization here.
+            if (!_isFunctionAppInitialized)
+            {
+                try
                 {
-                    FunctionLoader.SetupWellKnownPaths(functionLoadRequest);
-                    _powershellPool.Initialize(request.RequestId);
                     _isFunctionAppInitialized = true;
+                    InitializeForFunctionApp(request, response);
                 }
+                catch (Exception e)
+                {
+                    // Failure that happens during this step is terminating and we will need to return a failure response to
+                    // all subsequent 'FunctionLoadRequest'. Cache the exception so we can reuse it in future calls.
+                    _initTerminatingError = e;
```

Comment:
Note that, you don't need to set `IsDependencyDownloaded` to `false` here since it's `false` by default, unless the download succeeded and we it's set to `true`.

Reply:
That makes sense. I double checked with @SatishRanjan to make sure I am not missing anything when sending the response to the Host. Thanks.

Reply:
And, in fact, the `DependencyInstallationException` only covers part of the installation problem. For all exceptions thrown from the `dependencyManager.Initialize` are not warped by that exception. Maybe it should ...

Reply:
As per our conversation, now when an exception takes place in ```dependencyManager.Initialize```, we wrap it as a ```DependencyInstallationException ``.